pkg/conditions/conditions.go

```diff
@@ -0,0 +1,16 @@
+package conditions
+
+import (
+	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
+)
+
+const (
+	// BindingReady indicates that the binding succeeded
+	BindingReady conditionsv1.ConditionType = "Binding is ready"
+
+	// BindingInProgress indicates that binding is in progress
+	BindingInProgress conditionsv1.ConditionType = "Binding is in progress"
+
+	// BindingFailed indicates that the binding failed
+	BindingFailed conditionsv1.ConditionType = "Binding failed"
```

Comment:
It is not required to have a `Failed` type, since the negation of `Ready` is a failure regardless, this can go.